### PR TITLE
Add view, projection and aspect_ratio as camera properties

### LINKED ISSUE #7761: Allow directly retrieving "view" and "projection" from a built-in camera component, without making it active (DEF-2985)
**Is your feature request related to a problem? Please describe (REQUIRED):**
I want to use the built-in camera component to generate views/projections without having to make the camera "active" and wait a full update cycle. This would be a big step towards not needing to use addons (or custom code) to do anything non-basic with cameras.

**Describe the solution you'd like (REQUIRED):**
I would like, at a minimum: (or equivalent)
- `go.get("#camera", "view")`
- `go.get("#camera", "projection")`

The current API is also missing at least one other basic setting that you can set, but not get (according to the docs at https://defold.com/ref/stable/camera/) - I don't need this if the above are implemented, but it seems like a good idea to finish out the basic set/get API.
- `go.get("#camera", "aspect_ratio")`

**Describe alternatives you've considered (REQUIRED):**
I could probably simulate this functionality by reading the camera position, rotation, and settings, and then using `vmath.matrix4_perspective` and `vmath.matrix4_look_at`, but this is a lot more steps, requires me to know how to calculate the parameters for `matrix4_look_at`, and requires me to know, at a minimum, the existing `aspect_ratio` setting since I cannot read it.

Documenting this alternate process in the camera/vmath documentation would satisfy my immediate need, but I'm hoping this can be considered core functionality for the engine itself.

**Additional context (OPTIONAL):**
This is the 2nd task out of the 6 listed on https://github.com/defold/defold/issues/3227 and was already tagged as "DEF-2985" in that list.

### PR DESCRIPTION
**Pull request by britzl**
The view, projection and aspect ratio of a camera component are now available as properties on the component. The view and projection are read only using `go.get()`, while the aspect ratio can also be set using `go.set()`.

```lua
local view = go.get("#camera", "view")
local projection = go.get("#camera", "projection")
local aspect_ratio = go.get("#camera", "aspect_ratio")
go.set("#camera", "aspect_ratio", 1.4)
```

Fixes #7761 

## PR checklist

* [x] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [x] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [x] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: engine/gamesys/proto/gamesys/camera_ddf.proto [hunk 1]
```diff
@@ -170,3 +170,56 @@ message ReleaseCameraFocus {}
  * end
  * ```
  */
+
+/*# [type:float] camera projection
+ *
+ * [mark:READ ONLY] The calcualted projection matrix of the camera.
```
Comment (JCash) [inline engine/gamesys/proto/gamesys/camera_ddf.proto:176]: typo: calculated

File: engine/gamesys/proto/gamesys/camera_ddf.proto [hunk 2]
```diff
@@ -170,3 +170,56 @@ message ReleaseCameraFocus {}
...
+ *   local view = go.get("#camera", "view")
+ * end
+ * ```
+ */
+
+/*# [type:float] camera aspect ratio
+ *
+ * The aspect ratio of the camera. Used when calculating the
+ * projection of a perspective camera.
+ * The type of the property is number.
```
Comment (JCash) [inline engine/gamesys/proto/gamesys/camera_ddf.proto:212]: Perhaps better to describe what this ratio represents?
E.g. the relationship between the horizontal/vertical fov angle (or something)

File: engine/gamesys/src/gamesys/components/comp_camera.cpp [hunk 3]
```diff
@@ -356,6 +385,11 @@ namespace dmGameSystem
             component->m_OrthographicZoom = params.m_Value.m_Number;
             return dmGameObject::PROPERTY_RESULT_OK;
         }
+        else if (CAMERA_PROP_ASPECT_RATIO == set_property)
```
Comment (JCash) [inline engine/gamesys/src/gamesys/components/comp_camera.cpp:388]: I wonder if we should have a "READ_ONLY" return value 🤔 
"Not found" is a bit misleading

**Review by JCash — DISMISSED** [review]

File: engine/gamesys/src/gamesys/components/comp_camera.cpp (hunk 3 above)
Comment (britzl) [reply engine/gamesys/src/gamesys/components/comp_camera.cpp:388]: Yes, this is a good point. I'll add it.

Comment (britzl) [reply engine/gamesys/src/gamesys/components/comp_camera.cpp:388]: Added!

File: engine/gamesys/proto/gamesys/camera_ddf.proto (hunk 1 above)
Comment (britzl) [reply engine/gamesys/proto/gamesys/camera_ddf.proto]: Fixed

File: engine/gamesys/proto/gamesys/camera_ddf.proto (hunk 2 above)
Comment (britzl) [reply engine/gamesys/proto/gamesys/camera_ddf.proto]: Added!

File: engine/gameobject/src/dmsdk/gameobject/gameobject.h [hunk 4]
```diff
@@ -198,6 +199,7 @@ namespace dmGameObject
         PROPERTY_RESULT_RESOURCE_NOT_FOUND = -10,
         PROPERTY_RESULT_INVALID_INDEX = -11,
         PROPERTY_RESULT_INVALID_KEY = -12,
+        PROPERTY_RESULT_READ_ONLY = -13,
```
Comment (JCash) [inline engine/gameobject/src/dmsdk/gameobject/gameobject.h:202]: 👍

**Review by JCash — APPROVED** [review]

**Pull request merged**